app.py

```diff
@@ -69,10 +71,92 @@ def marks_api():
 
         return jsonify(serialize_db_mark(mark)), 201
     if request.method == "GET":
-        marks = Mark.select(Mark, Student).join(Student)
+        marks = Mark.select(Mark, Student).join(Student).join_from(Mark,Teacher)
 
         return jsonify([serialize_db_mark(mark) for mark in marks])
 
+@app.route('/teachers', methods=["GET", "POST"])
+def teachers_api():
+    if request.method == "POST":
+
+        data = deserialize_teacher_data()
+
+        validate_teacher_data(data)
+        # validated_data["student"] = student
+
+        teacher = Teacher.create(**data)
+        # mark.student = student
+
+        return jsonify(serialize_db_teachers(teacher)), 201
+    if request.method == "GET":
+        #marks = Mark.select(Mark, Student).join(Student)
+        teachers = Teacher.select()
+
+        return jsonify([serialize_db_teachers(teacher) for teacher in teachers])
+
+@app.route('/teachers/<int:teacher_id>', methods=["GET"])
+def teacher_api(teacher_id):
+    if request.method == "GET":
+        teacher = Teacher.get_or_none(id=teacher_id)
+
+        if not teacher:
+            return jsonify({"message": "teacher not found"}), 404
+
+        return jsonify(serialize_db_teacher_with_id(teacher))
+
+@app.route('/delteacher', methods=["DELETE"])
+def del_teacher_with_data():
+    if request.method == "DELETE":
+        data = deserialize_teacher_data()
+
+        if not data:
+            return jsonify({"error": "Can`t get teacher`s data"}), 400
+
+        teacher_to_delete = Teacher.get_or_none(Teacher.name == data["name"],Teacher.subject == data["subject"])
+
+        if teacher_to_delete:
+            teacher_to_delete.delete_instance()
+            return jsonify({"message": "Teacher has been deleted"}), 200
+        else:
+            return jsonify({"error": "The teacher not found"}), 404
+
+@app.route('/delteacher_id/<int:teacher_id>', methods=["DELETE"])
+def del_teacher_with_id(teacher_id):
+    if request.method == "DELETE":
+        teacher_to_delete = Teacher.get_or_none(id=teacher_id)
+        if teacher_to_delete:
+            teacher_to_delete.delete_instance()
+            return jsonify({"message": "Teacher has been deleted"}), 200
+        else:
+            return jsonify({"error": "The teacher not found"}), 404
+
+@app.route('/updateteacher/<teacher_id>', methods=["PATCH"])
```

Comment:
The URL should represent just resource name and not any action. Action is set by request method
```suggestion
@app.route('/teacher/<teacher_id>', methods=["PATCH"])
```